src/NHibernate/Proxy/DefaultDynamicProxyMethodCheckerExtensions.cs

```diff
@@ -23,12 +23,10 @@ public static bool IsProxiable(this MethodInfo method)
 		public static bool ShouldBeProxiable(this MethodInfo method)
 		{
 			// to use only for real methods (no getter/setter)
-			return (method.DeclaringType != typeof (MarshalByRefObject)) &&
-			       !IsFinalizeMethod(method) &&
```

Comment:
The `!IsFinalizeMethod(method)` check is not necessary, because object destructors are protected. This method only considers `public`, `internal` and `protected internal` methods.